src/GlobalMenuDrawer/index.js

```diff
@@ -79,37 +81,62 @@ const GlobalMenuDrawer = () => {
   const token = useSelector((state) => state.data.token);
   const user = useSelector((state) => state.data.user);
 
-  const modals = useMemo(() => [
-    ...(dailyReportEnabled ? [{
-      title: t('dailyReportModal.title'),
-      content: DailyReportModal,
-      modalProps: { className: 'daily-report-modal' },
-    }] : []),
-    {
-      title: t('fieldReportsModal.title'),
-      content: DataExportModal,
-      url: 'activity/events/export',
-      paramString: calcEventFilterForRequest(),
-      children: <div>{t('fieldReportsModal.content')}</div>
-    },
-    ...(kmlExportEnabled ? [{
-      title: t('masterKMLModal.title'),
-      content: KMLExportModal,
-      url: 'subjects/kml/root',
-      modalProps: { className: 'kml-export-modal' },
-    }] : []),
-    {
-      title: t('subjectInformationModal.title'),
-      content: DataExportModal,
-      url: 'trackingmetadata/export',
-    },
-    {
-      title: t('subjectReportsModal.title'),
-      content: DataExportModal,
-      url: 'trackingdata/export',
-    },
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  ], [dailyReportEnabled, eventFilter, eventTypes, kmlExportEnabled, t]);
+  const modals = useMemo(() => {
+    const modals = [
+    ];
+
+    if (dailyReportEnabled) {
+      modals.push(
+        {
+          title: t('dailyReportModal.title'),
+          content: DailyReportModal,
+          modalProps: { className: 'daily-report-modal' },
+        }
+      );
+    }
+
+    if (kmlExportEnabled) {
+      modals.push(
+        {
+          title: t('masterKMLModal.title'),
+          content: KMLExportModal,
+          url: 'subjects/kml/root',
+          modalProps: { className: 'kml-export-modal' },
+        }
+      );
+    }
+
+    if (hasObservationExportPermissions) {
+      modals.push(...[
+        {
+          title: t('subjectInformationModal.title'),
+          content: DataExportModal,
+          url: 'trackingmetadata/export',
+        },
+        {
+          title: t('subjectReportsModal.title'),
+          content: DataExportModal,
+          url: 'trackingdata/export',
+        }
+      ]);
+    }
+
+    if (hasEventExportPermissions) {
+      modals.push(
+        {
+          title: t('fieldReportsModal.title'),
+          content: DataExportModal,
+          url: 'activity/events/export',
+          paramString: calcEventFilterForRequest(),
+          children: <div>{t('fieldReportsModal.content')}</div>
+        }
+      );
+    }
+
+    return modals;
+    /* eslint-disable-next-line react-hooks/exhaustive-deps */
```

Comment:
Is there a reason for this eslint-disable?

Reply:
Yes -- I'm preserving this from the old code actually. There's a bit of a break in the paradigm of store-driven state changes here, we pass `eventFilter` in the dependency array to ensure the value returned by `calcEventFilterForRequest` always matches the current state of the event filter.